examples/structured_data/advanced_techniques_for_structured_data.py

```diff
@@ -0,0 +1,472 @@
+
+NUMERIC_FEATURE_NAMES = [
+    "Aspect",
+    "Elevation",
+    "Hillshade_3pm",
+    "Hillshade_9am",
+    "Hillshade_Noon",
+    "Horizontal_Distance_To_Fire_Points",
+    "Horizontal_Distance_To_Hydrology",
+    "Horizontal_Distance_To_Roadways",
+    "Slope",
+    "Vertical_Distance_To_Hydrology",
+]
+
+CATEGORICAL_FEATURES_WITH_VOCABULARY = {
+    "Soil_Type": list(data["Soil_Type"].unique()),
+    "Wilderness_Area": list(data["Wilderness_Area"].unique()),
+}
+
+CATEGORICAL_FEATURE_NAMES = list(CATEGORICAL_FEATURES_WITH_VOCABULARY.keys())
+
+FEATURE_NAMES = NUMERIC_FEATURE_NAMES + CATEGORICAL_FEATURE_NAMES
+
+COLUMN_DEFAULTS = [
+    [0] if feature_name in NUMERIC_FEATURE_NAMES + [TARGET_FEATURE_NAME] else ["NA"]
+    for feature_name in CSV_HEADER
+]
+
+NUM_CLASSES = len(TARGET_FEATURE_LABELS)
+
+"""
+Second, let's download the train and test data files
+
+We can now load the data in Pandas Dataframes. The training data split includes 431,010 samples,
+while the test data split includes 75,001 samples, with 13 columns per sample (12 features, plus the
+target label):
+"""
+
+"""
+## Experiment setup
+
+We create an input function to read and parse the file, and convert features and labels into a
+[`tf.data.Dataset`](https://www.tensorflow.org/guide/datasets) for training or evaluation.
+"""
+
+
+def get_dataset_from_csv(csv_file_path, batch_size, num_epochs=None, shuffle=False):
+
+    dataset = tf.data.experimental.make_csv_dataset(
+        csv_file_path,
+        batch_size=batch_size,
+        column_names=CSV_HEADER,
+        column_defaults=COLUMN_DEFAULTS,
+        label_name=TARGET_FEATURE_NAME,
+        num_epochs=num_epochs,
+        header=True,
+        shuffle=shuffle,
+    )
+
+    return dataset.cache()
+
+
+"""
+Here we configure the parameters and implement the procedure for running a training and evaluation
+experiment given a model.
+"""
+
+train_size = 493860
+learning_rate = 0.001
+dropout_rate = 0.1
+batch_size = 265
+num_epochs = 50
+
+train_steps_per_epoch = train_size // batch_size
+hidden_units = [32, 32]
+
+
+def run_experiment(model):
+
+    model.compile(
+        optimizer=keras.optimizers.Adam(learning_rate=learning_rate),
+        loss=keras.losses.SparseCategoricalCrossentropy(),
+        metrics=[keras.metrics.SparseCategoricalAccuracy()],
+    )
+
+    train_dataset = get_dataset_from_csv(train_data_file, batch_size, shuffle=True)
+
+    test_dataset = get_dataset_from_csv(test_data_file, batch_size, num_epochs=1)
+
+    print("Start training the model...")
+    history = model.fit(
+        train_dataset, epochs=num_epochs, steps_per_epoch=train_steps_per_epoch
+    )
+    print("Model training finished")
+
+    _, accuracy = model.evaluate(test_dataset, verbose=0)
+
+    print(f"Test accuracy: {round(accuracy * 100, 2)}%")
+
+
+"""
+## Create model inputs
+
+We define the inputs for our models as a dictionary, where the key is the feature name, and the value
```

Comment:
Suggestion, for more "variety" in the description of steps:

```suggestion
Now, define the inputs for the models as a dictionary, where the key is the feature name, and the value
```

(Not necessary, but good to know: second person ("you" - the reader vs first person ("we") https://developers.google.com/style/person)